bin/run

```diff
@@ -25,4 +25,6 @@ game_looper = GameLooper.new(ui:, board:, players:, outcome_checker:)
 
 ui.with_greeting_and_salutation do
   game_looper.loop
+  final_player = game_looper.current_player
```

Comment:
Ooh I like the idea of a result object! I was thinking of something along those lines but final_player = game_looper.loop didn't make sense to me. I like result = game_looper.loop much better, with the ability to access result.final_player. Currently, the game looper doesn't know the outcome (those details are hidden in outcome_checker), so it would probably just hold onto the last player in case of a win.

Reply:
@croese I'm a little stuck on how I might implement this. The Game Looper doesn't currently know the result of game, just that it's over and who the last player was. I played around with the idea of creating a Result object that contained the last player, and messages for the UI to display if the Outcome Checker found a win or a draw. I couldn't come up with a name for that class so I took that as an indication that it might not be cohesive enough. 

I still like the idea of a result object so if you have thoughts about how to implement it I'd love to hear!

Reply:
Sorry about the delay, I didn't see the notification of this reply for some reason.

I probably could've provided a little more detail in the initial comment. My thought was that the result object would have data that indicated the win/loss/draw/exited early status of the game loop's exit along with any player info that was relevant to that end state since there's no more computation to be done at that point. Then the display code could look at that data object's fields and output accordingly.

Another idea for situations like this is to use more of a push style, whenever the game loop hits an exit condition, it calls some method on another object like `report_win` or `report_loss` that takes whatever args that are needed to report that and that object displays it. This is similar to an event or callback where you setup a trigger for later but you don't necessarily know when it's going to get called since it depends on, in this case, the users' moves.

Reply:
One thing that could be nice about the result object would be having something that tracks all the moves and history and that could be passed around easily and it would all live in one place. It could also include some of the information that's currently stored in the board, like the move history.

Reply:
Thanks for the thoughtful comments, @graceolivia and @croese! 

It's helpful to hear a description of a couple of contrasting approaches. I've been going for more of a push style, and I think I'll have to do a slight variation on what you described. Since I'm currently hiding the knowledge of the game's outcome from GameLooper, I ended up placing a call to `ui.display_outcome` in bin/run which I believe may be doing something similar to what you described with `report_win` or `report_loss`. 

Doing it this way, I needed to call `outcome_checker.win?` and `outcome_checker.draw? ` in `UI#display_outcome`. I'm wondering if there's a way to avoid calls from UI to Outcome Checker without pushing all of the display logic over into Outcome Checker.